tests/client/api_v1/test_get_balance.c

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unity/unity.h>
+
+#include "client/api/v1/get_balance.h"
+#include "core/utils/byte_buffer.h"
+
+#define ADDR_HEX "7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006"
+#define ADDR_INV "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+
+void test_get_balance() {
+  iota_client_conf_t conf = {
+      .url = "http://0.0.0.0/",
+      .port = 14265  // use default port number
+  };
+
+  res_balance_t* res = res_balance_new();
+
+  // test null cases
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(NULL, NULL, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(NULL, NULL, res));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(&conf, NULL, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(&conf, NULL, res));
+
+  // test invalid address
+  TEST_ASSERT_EQUAL_INT(-1, get_balance(&conf, ADDR_INV, res));
+  TEST_ASSERT_EQUAL_STRING(
+      "bad request, error: invalid address: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, error: "
+      "encoding/hex: invalid byte: U+0078 'x': invalid parameter",
+      res->u.error->msg);
+
+  // reset res
+  res_balance_free(res);
+  res = res_balance_new();
```

Comment:
C doesn't set it to NULL after freeing the memory, you need to do it manually.
```suggestion
  // reset res
  res_balance_free(res);
  res = NULL;
  res = res_balance_new();
  TEST_ASSERT_NOT_NULL(res);
```